contracts/index_executor/src/storage.rs

```diff
@@ -0,0 +1,323 @@
+            .map(|byte| format!("{byte:02x}"))
+            .collect::<String>();
+        let api: String;
+
+        pink_extension::debug!("write_storage: id: {}", &key);
+
+        let cmd = format!(
+            r#"{{
+                "fields": {{
+                    "id": {{
+                      "stringValue": "{key}"
+                    }},
+                    "data": {{
+                      "stringValue": "{data_str}"
+                    }}
+                }}
+            }}"#
+        );
+        if storage_data.is_some() {
+            pink_extension::debug!("write_storage: storage item already exist in with document id: {}, update it if necessary", storage_data.clone().unwrap().1);
+            if &storage_data.clone().unwrap().0 == data {
+                pink_extension::debug!("write_storage: same storage data, ignore");
+                return Ok(());
+            }
+            api = format!("documents/index-storage/{}", storage_data.unwrap().1);
+            let _ = self.send_request("PATCH", &api[..], &cmd)?;
+        } else {
+            pink_extension::debug!(
+                "write_storage: storage item doesn't exist in storage, trying to create one"
+            );
+            api = "documents/index-storage".to_string();
+            let _ = self.send_request("POST", &api[..], &cmd)?;
+        }
+
+        Ok(())
+    }
+
+    /// Remove a document from remote storage
+    fn remove_storage_item(&self, key: &[u8]) -> Result<(), &'static str> {
+        let storage_data: Option<(Vec<u8>, String)> = self.read_storage(key)?;
+        if storage_data.is_some() {
+            let api = format!("documents/index-storage/{}", storage_data.unwrap().1);
+            let _ = self.send_request("DELETE", &api[..], "")?;
+        }
+        Ok(())
+    }
+
+    /// Put or update a storage item to the remote storage
+    pub fn put(&self, key: &[u8], data: &Vec<u8>) -> Result<(), &'static str> {
+        self.write_storage(key, data)
+    }
+
+    /// Get a storage item from remote storage
+    pub fn get(&self, key: &[u8]) -> Result<Vec<u8>, &'static str> {
+        if let Some((storage_data, _)) = self.read_storage(key)? {
+            return Ok(storage_data);
+        }
+        Err("StorageItemNotFound")
+    }
+
+    /// Delete a storage item from remote storage
+    pub fn delete(&self, key: &[u8]) -> Result<(), &'static str> {
+        self.remove_storage_item(key)
+    }
+
+    /// Upload task data to remote storage
+    pub fn upload_task(&self, task: &Task) -> Result<(), &'static str> {
+        self.write_storage(&task.id, &task.encode())
+    }
+
+    /// Lookup task data from remote storage, return None if not found
+    pub fn lookup_task(&self, id: &TaskId) -> Option<Task> {
+        if let Ok(Some((storage_data, _))) = self.read_storage(id) {
+            return match Decode::decode(&mut storage_data.as_slice()) {
+                Ok(task) => Some(task),
+                _ => None,
+            };
```

Comment:
Equivalent to `Decode::decode(...).ok()`. However, then you didn't handle the decode error. It makes future debug harder.

Same for the functions below

Reply:
Let's ignore in this PR, these two functions already removed in following PR